File: Client/src/app/items/models/Item.ts
import { IReview } from 'src/app/reviews/models/review';
import { User } from 'src/app/auth/models/user';
import {IItemCategory} from 'src/app/items/models/ItemCategory';

export class Item {

}

export interface IItem {
    name: string;
    title: string;
    description: string;
    price: number;
    id: string;
    createdDate: Date;
    updatedDate: Date;
    createdBy: User;
    updatedBy: User;
    reviews: IReview[];
    images: Array<File>;
    itemCategory: IItemCategory;
}

export interface IItemCreate {
    name: string;
    title: string;
    description: string;
    price: number;
    images: Array<File>;
}

export interface IItemDetails {
    name: string;
    title: string;
    description: string;
    price: number;
    reviews: Array<IReview>;
    images: Array<string>;
    createdBy: User;
    updatedBy: User;
}

File: WebServer/Models/Items/Item.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace WebServer.Models.Items
{
    public class Item : Base
    {
        public Item()
        {
            Images = new List<Image>();
            Reviews = new List<Review>();
        }
        public string Name { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }

        [DataType(DataType.Currency)]
        public double Price { get; set; }

        public bool Sold{get;set;}

        public virtual ItemCondition ItemCondition{get;set;}

        public virtual ItemCategory ItemCategory { get; set; }

        public virtual DeliveryOption DeliveryOption{get;set;}

        public virtual List<Review> Reviews { get; set; }

        public virtual List<Image> Images { get; set; }
    }
}
